Storage/ThirdParty/SwiftData.swift

```diff
@@ -92,10 +103,30 @@ open class SwiftData {
             if self.sharedConnection == nil {
                 log.debug(">>> Creating shared SQLiteDBConnection for \(self.filename) on thread \(Thread.current).")
                 self.sharedConnection = ConcreteSQLiteDBConnection(filename: self.filename, flags: SwiftData.Flags.readWriteCreate.toSQL(), key: self.key, prevKey: self.prevKey)
+                didCreateConnection = true
             }
             connection = self.sharedConnection
         }
 
+        // Should only be `true` when the `ConcreteSQLiteDBConnection` is first created. This
+        // happens when the database is accessed for the very first time. It should also happen
+        // again when the database is first accessed after being re-opened. Either way, this is
+        // the moment when we want to attempt to setup our connection (i.e. create the schema).
+        if didCreateConnection {
+            // If there is no callback for setting up the connection, it means we have already
+            // succeeded in setting it up and there is no need to re-run it.
```

Comment:
Comment no longer relevant. Consider whether you need `didCreateConnection` at all; why not simply put

```
let conn = ConcreteSQLiteDBConnection(…)
if !connectionCreatedCallback(self, conn) {
    self.closed = true
    return nil
} else {
    self.sharedConnection = conn
    return conn
}
```

up on line 106?